lazyscribe/repository.py

```diff
@@ -63,14 +64,20 @@ def __init__(
         self.dir = self.fpath.parent
         self.storage_options = storage_options
 
-        # If in ``r``, ``a``, or ``w+`` mode, read in the existing repository.
         self.artifacts: list[Artifact] = []
         self.fs = fsspec.filesystem(self.protocol, **storage_options)
 
         if mode not in ("r", "a", "w", "w+"):
             raise ValueError("Please provide a valid ``mode`` value.")
+        if mode == "a":
+            warnings.warn(
+                '`mode="a"` is deprecated and will be removed from `lazyscribe.repository.Repository` in version 2.0',
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            mode = "w+"
```

Comment:
Can you add a short test to verify that `mode` is being overwritten and the warning is being raised? See

https://github.com/lazyscribe/lazyscribe/blob/85904fb9efc2e0e7b45de4835d50443ff31fce4d/tests/test_repository.py#L329-L346

for an example.